test/test_grover.py

```diff
@@ -17,9 +17,9 @@
 from test import QiskitAlgorithmsTestCase
 
 import numpy as np
-from ddt import data, ddt
+from ddt import data, ddt, idata, unpack
 from qiskit import QuantumCircuit
-from qiskit.circuit.library import GroverOperator, PhaseOracle, PhaseOracleGate
+from qiskit.circuit.library import grover_operator, GroverOperator, PhaseOracle, PhaseOracleGate
```

Comment:
While `GroverOperator` is still supported -- which will be until Qiskit v3.0 -- we should test using the class and the function. We just need to think of capturing the deprecation warning when using the class

Reply:
If I'm not mistaken, the tests use both `grover_operator` and `GroverOperaor` when applicable (I've added an `oracle_builder` argument that can be either `grover_operator` or `GroverOperator`). Did I miss some places where we should test using both? 

Or are you talking about your previous comment, where the function should have a different behavior depending on the Qiskit version?

Maybe we could add in the CI a run using the oldest supported Qiskit version? And we could then duplicate the test, skipping one of the other depending on the Qiskit version. That would also allow to check whether the deprecation warning is emitted. WOuld that work for you?